src/main/proto/org/dependencytrack/notification/v1/notification.proto

```diff
@@ -216,6 +218,11 @@ message ProjectVulnAnalysisCompleteSubject {
   string token = 4;
 }
 
+message UserPrincipalSubject {
```

Comment:
Nit: Consider just calling it `UserSubject`. The `Principal` part is more of a [Java-specific thing](https://www.baeldung.com/security-context-basics) that IMO should not "bleed" into our public APIs.